core/cas-server-core-authentication-mfa-api/src/main/java/org/apereo/cas/authentication/DefaultMultifactorAuthenticationContextValidator.java

```diff
@@ -75,18 +75,6 @@
             LOGGER.debug("Requested authentication context [{}] is satisfied since device is already trusted", requestedContext);
             return Pair.of(Boolean.TRUE, requestedProvider);
         }
-        if (attributes.containsKey(MultifactorAuthenticationProviderBypass.AUTHENTICATION_ATTRIBUTE_BYPASS_MFA)
```

Comment:
Why this?

Reply:
This resolves cases where one service may have a bypass defined in the Service Registry.  If that service is used for primary login and a second service that requires MFA is requested as part of an SSO, then the user is not stopped and challenged for MFA.  The DefaultRequestedAuthenticatioContextValidator was changed to evaluate Bypass and Failure modes, just like it checks for MFA trigger on each request.